ApollosProject/apollos-shovel
# Crossroads Algolia Indexes

### About

This PR creates an algolia index for Crossroads Kids Club and adds the indexing step to both the backfill dag and the sync dag.

I largely copied the code from `dags/algolia_multitenant/algolia_content_dag.py`, and made changes necessary for crossroads. The following changes are relevant:

- We're using Crossroads' Algolia instance instead of the cluster's. Crossroads pays for a fancy algolia account and I see us wanting to use some of those features in the future. 
- We're using the description for the summary field in algolia. To do so, I had to remove the HTML markup from the description field using the new `HTMLParser` class.

### Testing
1. Run the backfill DAG for kids club. Verify that the algolia indexing step completes.
2. Check out the results in algolia's dashboard (I'm happy to give you the credentials for this account) and verify everything looks good.

### Screenshots
|Backfill|Sync|
|---|---|
|<img width="380" alt="Screen Shot 2022-02-07 at 3 07 36 PM" src="https://user-images.githubusercontent.com/34075293/152863768-7032f2ed-f462-4997-b9e7-8f696798904d.png">|<img width="618" alt="Screen Shot 2022-02-07 at 2 58 06 PM" src="https://user-images.githubusercontent.com/34075293/152863749-adb94d2c-1bdd-4f7b-a4f6-c18eb86b2330.png">|
